src/dataset-builder/DatasetBuilderUtils.ts

```diff
@@ -114,54 +69,60 @@ export interface Cohort extends DatasetBuilderType {
   criteriaGroups: CriteriaGroup[];
 }
 
-export interface DatasetBuilderType {
-  name: string;
-}
-
 export type DatasetBuilderValue = DatasetBuilderType;
 
 export type ValueSet = {
   domain: string;
   values: DatasetBuilderValue[];
 };
 
-export interface GetConceptsResponse {
-  result: Concept[];
-}
-
-export interface SnapshotBuilderParentConcept {
-  parentId: number;
-  children: Concept[];
-}
-
-export interface GetConceptHierarchyResponse {
-  readonly result: SnapshotBuilderParentConcept[];
-}
-
-export interface SearchConceptsResponse {
-  result: Concept[];
-}
-
 export type DatasetRequest = {
   cohorts: Cohort[];
-  conceptSets: ConceptSet[];
+  conceptSets: SnapshotBuilderDatasetConceptSet[];
   valueSets: ValueSet[];
 };
 
-export type DatasetAccessRequest = {
+export type SnapshotAccessRequest = {
   name: string;
   researchPurposeStatement: string;
   datasetRequest: DatasetRequest;
 };
 
-export const convertValueSet = (valueSet: ValueSet): ValueSetApi => {
+export type SnapshotBuilderCountRequest = {
+  cohorts: Cohort[];
+};
+
+export interface DomainOption extends SnapshotBuilderOption {
+  kind: 'domain';
+  root: SnapshotBuilderConcept;
+  conceptCount?: number;
```

Comment:
what does concept count represent? The number of concepts or the number of participants

Reply:
This may be wrong, but I believe `conceptCount` represents the amount of concepts that are in the DomainOption

Reply:
I believe it's the number of concepts in the domain that have at least one entry in that domain's occurrence table (e.g., there's at least one `person` that maps to that concept).